app/controllers/hospitals_controller.rb

```diff
@@ -2,4 +2,22 @@ class HospitalsController < ApplicationController
   def index
     @hospitals = Hospital.all
   end
+
+  def create
+    unless Hospital.exists?(hospital_params)
```

Comment:
Oh, I just spotted this. What is this doing for you? Isn't the `else` block handling the error?